### PR TITLE
Update `controller-tools` from 0.13.0 to 0.14.0

### PR DESCRIPTION
**Pull request by bayandin**
This PR updates `controller-tools` to 0.14.0.

`controller-tools` 0.13 segfauls with `go` ≥ 1.22, which blocks upgrading `go` (see https://github.com/kubernetes-sigs/controller-tools/issues/880).

Extracted from https://github.com/neondatabase/autoscaling/pull/1035

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -410,7 +410,7 @@ ENVTEST_K8S_VERSION = 1.28.3
 
 CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
 # k8s deps @ 1.28.0 https://github.com/kubernetes-sigs/controller-tools
-CONTROLLER_TOOLS_VERSION ?= v0.13.0
+CONTROLLER_TOOLS_VERSION ?= v0.14.0
```
Comment (sharnoff) [inline Makefile:413]: controller-tools v0.14.0 has k8s deps at 1.29.

Not sure what issues we might have there, if any.

Comment (bayandin) [reply Makefile:417]: I've added a more detailed comment regarding `CONTROLLER_TOOLS_VERSION`.

> but worth being cautious about..
We'll see on staging if it works 😉

Comment (sharnoff) [reply Makefile:417]: Haha - sounds good, thanks. I'll put images from this PR on staging today for testing, then approve assuming all looks good :)

Comment (bayandin) [reply Makefile:417]: Thanks!

I thought we just merge it and deploy the next version to staging 🤣 
But yeah, deploying from the branch sounds even better!

Comment (sharnoff) [reply Makefile:425]: Tested it on staging -- looks good!

**Review by sharnoff — APPROVED** [review]

**Pull request merged**